File: TOWebContentViewControllerExample/ViewController.m
//
//  ViewController.m
//  TOWebContentViewControllerExample
//
//  Created by Tim Oliver on 14/10/18.
//  Copyright © 2018 Tim Oliver. All rights reserved.
//

#import "ViewController.h"
#import "TOWebContentViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (IBAction)didTapButton:(id)sender
{
    // Get resources folder URL
    NSURL *resources = [[NSBundle mainBundle] resourceURL];
    NSURL *baseURL = [resources URLByAppendingPathComponent:@"HTML"];
    NSURL *fileURL = [baseURL URLByAppendingPathComponent:@"about.html"];

    TOWebContentViewController *webContentController = [[TOWebContentViewController alloc] initWithFileURL:fileURL baseURL:baseURL];
    webContentController.setsTitleFromContent = YES;

    UINavigationController *navigationController = [[UINavigationController alloc] initWithRootViewController:webContentController];
    navigationController.navigationBar.barStyle = UIBarStyleBlack;
    navigationController.modalPresentationStyle = UIModalPresentationFormSheet;
    navigationController.view.tintColor = [UIColor colorWithRed:91.0f/255.0f green:158.0f/255.0f blue:1.0f alpha:1.0f];
    [self presentViewController:navigationController animated:YES completion:nil];

    webContentController.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemDone target:self action:@selector(webContentViewControllerDone:)];
}

- (IBAction)webContentViewControllerDone:(id)sender
{
    [self dismissViewControllerAnimated:YES completion:nil];
}

@end

File: WebContentViewControllerExample/ViewController.swift
//
//  ViewController.swift
//  WebContentViewControllerExample
//
//  Created by Tim Oliver on 3/11/18.
//  Copyright © 2018 Tim Oliver. All rights reserved.
//

import UIKit

class ViewController: UIViewController, WebContentViewControllerDelegate {

    @IBAction func didTapLocalButton(sender: AnyObject) {
        // Get resources folder URL
        let resourcesURL = Bundle.main.resourceURL!
        let baseURL = resourcesURL.appendingPathComponent("HTML")
        let fileURL = baseURL.appendingPathComponent("about.html")

        let webContentController = WebContentViewController(fileURL: fileURL, baseURL: baseURL)
        webContentController.templateTags = ["AppName": "iComics"]
        webContentController.setsTitleFromContent = true
        webContentController.delegate = self;

        let navigationController = UINavigationController(rootViewController: webContentController)
        navigationController.navigationBar.barStyle = .black
        navigationController.modalPresentationStyle = .formSheet
        navigationController.view.tintColor = UIColor(red: 91.0/255.0, green: 158.0/255.0, blue: 1.0, alpha: 1.0)
        present(navigationController, animated: true, completion: nil)

        webContentController.navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(didTapDone(sender:)))
    }

    @IBAction func didTapWebButton(sender: AnyObject) {

    }

    @objc func didTapDone(sender: AnyObject) {
        dismiss(animated: true, completion: nil)
    }
}

